xml_converter/src/xml_converter.cpp

```diff
@@ -431,49 +431,77 @@ void write_taco_directory(string directory, map<string, Category>* marker_catego
     write_xml_file(xml_filepath, marker_categories, parsed_pois);
 }
 
-void convert_taco_directory(string directory, string output_directory, map<string, Category>* marker_categories, vector<Parseable*>* parsed_pois) {
-    move_supplementary_files(directory, output_directory);
-    read_taco_directory(directory, marker_categories, parsed_pois);
-}
-
 ////////////////////////////////////////////////////////////////////////////////
 // process_data
 //
 // The universal entrypoint into the xml converter functionality. Both the CLI
 // and the library entrypoints direct here to do their actual processing.
 ////////////////////////////////////////////////////////////////////////////////
 void process_data(
-    vector<string> input_paths,
-    string output_directory) {
-    auto begin = chrono::high_resolution_clock::now();
-
+    vector<string> input_taco_paths,
+    vector<string> input_waypoint_paths,
+
+    // These will eventually have additional arguments for each output path to
+    // allow for splitting out a single markerpack
+    vector<string> output_taco_paths,
+    vector<string> output_waypoint_paths,
+
+    // This is a special output path used for burrito internal use that splits
+    // the waypoint protobins by map id.
+    string output_split_waypoint_dir) {
+    // All of the loaded pois and categories
     vector<Parseable*> parsed_pois;
     map<string, Category> marker_categories;
 
-    try {
```

Comment:
Can you explain why this try is being removed? There are cases when we want the program to finish processing all of the folders when one of the folders gives us an error.

Reply:
Summarizing offline discussion to this point.
* This was no longer needed because the marker pack directory used to be hardcoded, meaning that all marker packs in the folder were always parsed. Now it is dynamic, meaning that a user who specified a marker pack that is causing errors would just need to remove the specific marker pack from the command arguments.
* This however still runs across an issue when a user is providing an folder such as `./markerpacks/*'. zsh will automatically expand these arguments but bash will not. Meaning that a bash user would be in a similar situation as they were before, except now instead of just removing a marker pack listing they would have to add all of the other marker packs manually.
* But, to a real user, not a developer. Is there any real utility in creating an output from a partial marker pack, or partial set of marker packs? In general we should not be throwing any errors that need to be caught if we can help it. And if an uncaught error reaches this point should we really create a partial markerpack output, split at a possibly arbitrary point in the processing? I would argue it is better to output nothing, alongside a loud error, then quietly output partially corrupted data.

Reply:
I hadn't considered the partial data. Thank you for clarifying. The thing about bash vs zsh is frustrating but not something most people will run into.